Make SVCB parameter "oblivious-gateway" (#21)
tfpauly/draft-ohai-svcb-config: draft-pauly-ohai-svcb-config.md

--- abstract

This document defines a parameter that can be included in SVCB and HTTPS
DNS resource records to denote that a service is accessible using Oblivious
HTTP, with an indication of which Oblivious Gateway Resource to use to access

> the service (aa an Oblivious Target Resource). This document also defines

Reviewer: Remove "aa" in the above line.

Author: ```suggestion
the service (an Oblivious Target Resource). This document also defines
```

> the service (aa an Oblivious Target Resource). This document also defines

Reviewer 2: ```suggestion
the service (as an Oblivious Target Resource). This document also defines
```

mechanisms to learn more details about the related Oblivious Gateway Resource,
such as its key configuration.

--- middle

# Introduction

Oblivious HTTP {{!OHTTP=I-D.draft-ietf-ohai-ohttp}} allows clients to encrypt
messages exchanged with an Oblivious Target Resource (target). The messages
are encapsulated in encrypted messages to an Oblivious Gateway Resource

> (gateway), which gates access to the target. The gateway is access via an

Reviewer: Replace "access" with "accessed"

in such a way that the relay cannot inspect the contents of messages, and
the gateway and target cannot discover the client's identity.

Since Oblivious HTTP deployments will often involve very specific coordination
between clients, relays, and gateways, the key configuration can often be
shared in a bespoke fashion. However, some deployments involve clients
discovering oblivious targets and their assoicated gateways more dynamically.
For example, a network may want to advertise a DNS resolver that is accessible
over Oblivious HTTP and applies local network resolution policies via mechanisms
like Discovery of Designated Resolvers ({{!DDR=I-D.draft-ietf-add-ddr}}. Clients
can work with trusted relays to access these gateways.

This document defines a mechanism to advertise that an HTTP service supports
Oblivious HTTP using DNS records, as a parameter that can be included in SVCB
and HTTPS DNS resource records {{!SVCB=I-D.draft-ietf-dnsop-svcb-https}}.
The presence of this parameter indicates that a service can act as an oblivious
target, and indicates an oblivious gateway that can provide access to the target.

This document also defines two well-known URIs {{!RFC8615}}, which

> are access on the oblivious gateway indicated in the SVCB record.

Reviewer: Replace "access" with "accessed"

Author: ```suggestion
are accessed on the oblivious gateway indicated in the SVCB record.
```

> are access on the oblivious gateway indicated in the SVCB record.

Reviewer 2: ```suggestion
are accessed on the oblivious gateway indicated in the SVCB record.
```

- "oblivious-configs", which can be used to look up key configurations
on a host that has been identified as an oblivious gateway using SVCB
records ({{well-known-config}}); and,

- "oblivious-gateway", which can be used to send oblivious gateway requests
to a host that has been identified as an oblivious gateway using SVCB
records ({{well-known-gateway}}).

This mechanism does not aid in the discovery of oblivious relays;

> the configuration of relays is out of scope for this document.

Reviewer 2: ```suggestion
relay configuration is out of scope for this document.
```

# Oblivious Gateway Resource Well-Known URI {#well-known-gateway}

Oblivious gateways that are advertised via SVCB or HTTPS records
need to receive gateway requests (sent from oblivious relays) on
a particular path. While oblivious gateways in general do not
require fixed or well-known paths, since they can use ad-hoc
configurations with clients and relays, gateways discovered using
the method described in this document need to offer a well-known
path.

Offering a single, well-known URI for each discovered gateway name allows
for simpler lookup of associated key configurations, simpler communication
between clients and relays, and makes client-targeting attacks more
difficult to execute. See {{security}} for more discussion.

This document defines a well-known URI {{!RFC8615}}, "oblivious-gateway",
which an oblivious gateway uses to receive gateway requests.

The URI is constructed using the name in the "oblivious-gateway"
SVCB parameter.

For example, the oblivious request URI for the following record:

~~~
svc.example.com. 7200  IN HTTPS 1 . (
     alpn=h2 oblivious-gateway=osvc.example.com )
~~~

would be "https://osvc.example.com/.well-known/oblivious-gateway".

> Request to this resource are expected to use the content type

Reviewer 2: ```suggestion
Requests to this resource are expected to use the content type
```

targeting attack. Oblivious gateways that are detected to use targeted
key configurations per-client MUST NOT be used.

Oblivious gateways that are accessed based on SVCB discovery are required
to offer a well-known oblivious gateway request path. This is done in
part to make client-targeting attacks more difficult. If the gateway request
path is communicated in the SVCB parameters, individual clients could
receive unique paths which could be used to identify them upon making
requests via a relay. While it is still possible for different clients to
receive different gateway names, this requires provisioning more gateway
names (which has overhead in DNS configuration and TLS certificate

> generation), and allows relays to more easily enforce allow-lists of

Reviewer 2: I might also add that this forces the existence of these gateway names to show up in places like CT, which means the malicious target can't deny the fact that they tried to target clients after the fact.

Author: Agreed, but moved to a GET on the gateway path for now

> known gateway names without needing to also check request paths.

Reviewer: The above attack of different gateway names is possible with an malicious target aiming to identify clients. The attack can be mitigated by the client by using an pre-configured encrypted resolver to retrieve the SVCB records. The client can then compare this against the records retrieved using DNR/DDR to identify any targeted attack.

Author: This is changed now to not use a well-known

Author: This is changed now to not use a well-known
